pkg/runtime/wfengine/wfengine_test.go

```diff
@@ -587,9 +596,9 @@ func TestPauseResumeWorkflow(t *testing.T) {
 	}
 }
 
-func startEngine(ctx context.Context, r *task.TaskRegistry) (backend.TaskHubClient, *wfengine.WorkflowEngine) {
+func startEngine(ctx context.Context, t *testing.T, r *task.TaskRegistry) (backend.TaskHubClient, *wfengine.WorkflowEngine) {
```

Comment:
is there a proper way to stop engine? If so please add code for that too. 
Are there race conditions in these tests anywhere? If not add them to the list of packages that do not have race conditions.

Reply:
if this requires more work create an issue and create a separate PR for the same.

Reply:
I have created https://github.com/dapr/dapr/issues/6264 as it is a wider issue, separate from current PR.